### DISCUSSION TITLE
Loading a preferredView="text" model in OMEdit leads to model alteration

### DISCUSSION THREAD

**Discussion by tburch-LSS** [OpenModelica/OpenModelica#12949] [Q&A]
I wasn't sure if I should open this as an issue... but I noticed that whenever I load and then open a model in OMEdit and that model contains annotation(preferredView="text"), the model automatically becomes altered (asterisk appears in top of tab and one is queried about saving before exiting) in some way which I cannot discern.

I am using the latest version: 1.24.0~dev.beta.0.

... and oddly enough, this only seems to happen when said model is the first thing opened in a session... :-/

**Comment by adeas31** [comment]
I just tested this with a model,

```
model M
annotation(preferredView = "text");
end M;
```

Works as expected. The model is not marked as unsaved/changed. Do you have a test model that we can use to reproduce the issue.

**Reply by tburch-LSS** [reply]
i just tested with your above model and it still occurs. a few more details: it happens when said model is the first thing to be opened in a session; i am using ubuntu 22.04.

**Reply by adeas31** [reply]
I tested this on Windows and WSL.

@perost can you please test and see what happens.

**Reply by perost** [reply]
The model isn't marked as changed when I open it on Linux either.

**Reply by tburch-LSS** [reply]
did you save the model, leave the session, and then restart and re-open the model? if so, then i cannot explain the difference in behavior... :-/

**Reply by perost** [reply]
I just saved the model in a file and opened it with OMEdit.

Does it only happen if the model has a `preferredView="text"` annotation, or does it also happen if you open any model and open the text view?

**Reply by tburch-LSS** [reply]
it only happens with the preferredView="text" annotation and only when i open said model as the first thing in a session.

**Reply by tburch-LSS** [reply]
just to be perfectly clear, perhaps i should also mention that the model itself is not actually altered. i checked by saving it as a new file and the new and old are identical. it's just that OMEdit behaves as if something has changed.

**Reply by adeas31** [reply]
Unfortunately, we can't reproduce the issue on our side. We might need more information about it.

Please make a screencast, if possible. You can also try to reset OMEdit settings. Maybe some setting is causing the issue. Close OMEdit, find `omedit.ini` and rename it to `omedit-O.ini`, start OMEdit.

**Reply by tburch-LSS** [reply]
i tried the omedit.ini renaming approach, but the problem (albeit a minor one) persists. i updated to the new 1.24.0 version and it persists there also. once again, this is on Ubuntu 22.04.5 LTS. the only other criteria is that the model with the  annotation(preferredView="text") is saved in a file and it is the first thing opened in an OMEdit session. i tried with and without having a Modelica library loaded first. this makes no difference. and again, the file itself is not actually altered. OMEdit only behaves as if this is so.

i will look into how to create a "screencast" and get back soon...

**Reply by tburch-LSS** [reply]
[Screencast from 10.10.2024 16:19:13.webm](https://github.com/user-attachments/assets/32efb829-f321-447c-8fea-f7f0d8ddc355)

**Reply by tburch-LSS** [reply]
i dare say that this is still an issue and has become worse after updating to Ubuntu 24.04 and OMEdit 1.26.3.
now the phenomenon happens whenever i switch to Text View on any model and whenever i re-expand OMEdit from being minimized. the problem comes with the follow scripting notification: 
"qt.qpa.wayland: Wayland does not support QWindow::requestActivate()"

<img width="1366" height="736" alt="image" src="https://github.com/user-attachments/assets/16adff60-c55d-43ce-b82b-abec856869e9" />

**Reply by tburch-LSS** [reply]
problem persists even with X11/Xorg windowing system... :-/
